### ISSUE TITLE
imap function

### ISSUE THREAD

**Issue by ahmetertem** [barbushin/php-imap#214]
I want to ask why `imap_utf7_encode` ing arguments on line 813 ? If password contains `&` character it's not getting authenticated. If i comment that line everything works fine. So, why we need to encode arguments ?

**Comment by cjprinse** [comment]
This is also a problem for imap_rfc822_parse_headers

**Comment by karakum** [comment]
Also this is why searching by utf-8 subject don't work properly. If comment out this encoding it does.
This is my workaround:
```
// $cfg['creteria'] = 'UNSEEN FROM noreply@mail.ru SUBJECT "Отчет по продажам"'
$mailsIds = imap_search($mailbox->getImapStream(), $cfg['criteria'], SE_UID, $mailbox->getServerEncoding());
```

**Comment by tweecool** [comment]
Having the same issue, "IMAP method imap_open() failed with error: LOGIN bad username or password". @ahmetertem is correct and works fine when you comment imap_utf7_encode line 847 with php-imap/php-imap (3.0.6)

**Comment by Sebi94nbg** [comment]
@barbushin can be closed as well: [Commit a6b2dd5](https://github.com/barbushin/php-imap/pull/296/commits/a6b2dd5409ed3de62696a1705fd27fedcfda06d7)

See PR #296.
